relayer/relayer/src/contracts/eth.rs

```diff
@@ -1,4 +1,61 @@
-use ethers::contract::abigen;
+use aleph_client::sp_core::H160;
+use ethers::{
+    contract::{abigen, ContractError},
+    middleware::Middleware,
+    prelude::BlockNumber,
+};
 
 abigen!(Most, "../eth/artifacts/contracts/Most.sol/Most.json");
 abigen!(WETH9, "../eth/artifacts/contracts/WETH9.sol/WETH9.json");
+
+pub enum SignatureState {
+    #[cfg_attr(feature = "evm", allow(dead_code))]
+    SignedNotFinalized,
```

Comment:
Alternatively, we could disable this variant totally when `evm` feature is enabled: this could also help us find all the places where this variant is used, but I imagine this could make test code more complicated in some cases. wdyt?

Reply:
I have refactored it a bit to solve this issue

Reply:
Not exactly what I meant: this still doesn't prevent us from accidentally using `Signed {finalized = false }` with `evm` feature, but it's not that important